### ISSUE TITLE
Using `@sanity/types` into Nuxt project introduces warnings

### ISSUE THREAD

**Issue by pascalwengerter** [sanity-io/sanity#4247]
**Describe the bug**

We face type errors when using `@sanity/types` in a Nuxt application because `@sanity/types` brings `@types/react` into the project.

**To Reproduce**

1. Use Sanity with `@sanity/types` && Nuxt3 in a project
2. Define a `class=""` within a Nuxt/Vue component template
3. See error, example screenshot below

**Expected behavior**

No issues or type warnings

**Screenshots**

![image (2)](https://user-images.githubusercontent.com/16822008/223469616-512923a5-74d2-4e33-aba3-56f94ddf2beb.png)

**Which versions of Sanity are you using?**

```
@sanity/cli (global)  3.5.0 (latest: 3.5.1)
@sanity/image-url     1.0.2 (up to date)
@sanity/types         3.4.0 (latest: 3.5.1)
```

**Which versions of Node.js / npm are you running?**

`v16.18.0`

**Additional context**

Relevant discussion upstream in Vue.js: https://github.com/vuejs/language-tools/discussions/592

**Comment by issenn** [comment]
Same problem.

**Comment by pascalwengerter** [comment]
Seems to have been resolved since updating Nuxt `v3.5.x` and Vue `v3.3.x` (also see https://github.com/vuejs/language-tools/discussions/592)
